ocaml/idl/datamodel_common.ml

```diff
@@ -679,7 +679,11 @@ let field ?(in_oss_since = Some "3.0.3") ?in_product_since
   (* in_product_since currently defaults to 'Some rel_rio', for backwards compatibility.
      	 * This should eventually become 'None'. *)
   let in_product_since =
-    match in_product_since with None -> Some rel_rio | x -> x
+    match in_product_since with
+    | None ->
```

Comment:
This is always evaluated, before the rule below, therefore in_product_since=None + lifecycle:Some would still raise, which I think was not the intention.

Reply:
(This is removed in the second commit, which wasn't pushed yet then)

Reply:
We should probably:
* either introduce a `rel_next` and require that when lifecycle is set the in_product_since is one of `rel_nile`, `rel_nile_preview`, or `rel_next`
* or (probably smaller and easier) is to keep allowing `in_product_since = None`, but only when `lifecycle` is specified. When `lifecycle` is not specific you must explicitly add `rel_rio`

We should also reject `rel_rio` with a lifecycle specified, which is likely wrong.